js/signaling.js

```diff
@@ -99,57 +131,119 @@
-				// to do the following:
-				//
-				// 1. Get a list of connected clients to the room
-				// 2. Return the list of connected clients
-				// 3. Connect to the room with the clients as list here
-				//
-				// The clients will then use the message command to exchange
-				// their signalling information.
-				var callback = arguments[2];
-				$.ajax({
-					url: OC.linkToOCS('apps/spreed/api/v1/call', 2) + data,
-					type: 'POST',
-					beforeSend: function (request) {
-						request.setRequestHeader('Accept', 'application/json');
-					},
-					success: function (result) {
-						this.sessionId = result.ocs.data.sessionId;
-						OCA.SpreedMe.Rooms.peers(data).then(function(result) {
-							var roomDescription = {
-								'clients': {}
+	InternalSignaling.prototype.joinCall = function(token, callback) {
+		// The client is joining a new room, in this case we need
+		// to do the following:
+		//
+		// 1. Get a list of connected clients to the room
+		// 2. Return the list of connected clients
+		// 3. Connect to the room with the clients as list here
+		//
+		// The clients will then use the message command to exchange
+		// their signalling information.
+		$.ajax({
+			url: OC.linkToOCS('apps/spreed/api/v1/call', 2) + token,
+			type: 'POST',
+			beforeSend: function (request) {
+				request.setRequestHeader('Accept', 'application/json');
+			},
+			success: function (result) {
+				console.log("Joined", result);
+				this.sessionId = result.ocs.data.sessionId;
+				this.currentCallToken = token;
+				this._startPingRoom();
+				this._getRoomPeers(token).then(function(result) {
+					var roomDescription = {
+						'clients': {}
+					};
+
+					result.ocs.data.forEach(function(element) {
+						if (this.sessionId !== element['sessionId']) {
+							roomDescription['clients'][element['sessionId']] = {
+								'video': true
 							};
+						}
+					}.bind(this));
+					callback('', roomDescription);
+				}.bind(this));
+			}.bind(this)
+		});
+	};
 
-							result.ocs.data.forEach(function(element) {
-								if (this.sessionId !== element['sessionId']) {
-									roomDescription['clients'][element['sessionId']] = {
-										'video': true
-									};
-								}
-							}.bind(this));
-							callback('', roomDescription);
-						}.bind(this));
-					}.bind(this)
-				});
-				break;
-			case 'message':
-				if(data.type === 'answer') {
-					console.log("ANSWER", data);
-				} else if(data.type === 'offer') {
-					console.log("OFFER", data);
+	InternalSignaling.prototype.leaveCall = function() {
+		this._stopPingRoom();
+		if (this.currentCallToken) {
+			$.ajax({
+				url: OC.linkToOCS('apps/spreed/api/v1/room', 2) + this.currentCallToken + '/participants/self',
+				type: 'DELETE'
+			});
+			this.currentCallToken = null;
+		}
+	};
+
+	InternalSignaling.prototype.leaveAllRooms = function() {
+		$.ajax({
+			url: OC.linkToOCS('apps/spreed/api/v1/call', 2) + this.currentCallToken,
```

Comment:
since it now has the token and is not room related but call, it should be `leaveCall` not `leaveAllRooms`